aips/aip-51.md

```diff
@@ -0,0 +1,65 @@
+---
+
+aip: 51
+
+title: Changing beneficiary addresses
+
+author: junkil-park, michelle-aptos, movekevin, wintertoro
+
+discussions-to: https://github.com/aptos-foundation/AIPs/issues/251
+
+Status: In Review
+
+type: Standard Framework
+
+created: 10/3/2023
+
+---
+
+
+# AIP-51 - Changing beneficiary addresses
+
+## Summary
+
+Right now, the operator commission goes to the operator's address. We have received multiple requests for the ability to set a different beneficiary address. One example use case is when operators want their commission to go to a cold storage wallet while their operator address is used for day-to-day operations. This feature should apply to both the staking_contract and the delegated_staking contract.
+
+
+### Goals
+
+The goal is to allow operators to set a different beneficiary address for their commission rewards. So, the following scenarios should be possible:
+- operators want to store commission rewards in a cold wallet. This change would introduce the decoupling of the wallet and operator keys so that funds can be sent to an address different from the operator's account.
+- The funds would remain in a cold wallet while the operator can perform other operations in a hot wallet.
+
+
+### Out of Scope
+
+Changing the beneficiaries for token owners is out of the scope of this AIP.
+
+
+### Impact
+
+This feature impacts operators and their beneficiaries.
+
+
+## Reference Implementation
+TBD
+
+## Testing
+- What is the testing plan?
+Automated unit tests and testing in devnet/testnet before releasing to mainnet.
+
+## Risks and Drawbacks
+Although the operator address is set to a new address, the beneficiary address could remain in the old one. The commission rewards can go to the wrong beneficiary address in this case. To prevent this, when the operator address is reset, the beneficiary should be set to the operator address.
```

Comment:
how fast do we expect this to take effect? Is it going to be updated immediately? or it would take an epoch?

Reply:
The beneficiary change takes effect immediately. The rewards will go to the new beneficiary from the subsequent reward distribution.

Only the operator can set its beneficiary. We assume that the operator account and the beneficiary account are under the control of the operator. So, it should be fine. Please let me know if you have any concerns.